### PR TITLE
fix(emotion): Support dynamic options

### PR DESCRIPTION
**Pull request by ImBrek**
```
const ContainerWithOptions = styled('div', makeOptions())`
  color: hotpink;
`
```

transforms to

```
const ContainerWithOptions = /*#__PURE__*/ styled('div', makeOptions(), {
    target: "e1imlkhz0",
    label: "ContainerWithOptions"
})("color:hotpink");
```

I think it is wrong. Third param should extend result of `makeOptions()`

### CODE REVIEW

File: packages/emotion/transform/src/lib.rs [hunk 1]
```diff
@@ -609,16 +625,32 @@ impl<C: Comments> Fold for EmotionTransformer<C> {
...
-                                            Expr::Object(ObjectLit {
+                                    match expr.as_mut() {
+                                        Expr::Object(ObjectLit { props, .. }) => {
+                                            props.extend(object_props);
+                                        }
+                                        Expr::Call(_) => {
+                                            object_props.push(PropOrSpread::Spread(
+                                                SpreadElement {
+                                                    dot3_token: DUMMY_SP,
+                                                    expr:expr.to_owned()
```
Comment (kdy1) [inline packages/emotion/transform/src/lib.rs:636]: This looks like a needless clone.

Comment (kdy1) [reply packages/emotion/transform/src/lib.rs:636]: `to_owned` copies the data

File: packages/emotion/transform/src/lib.rs [hunk 2]
```diff
@@ -482,17 +483,32 @@ impl<C: Comments> Fold for EmotionTransformer<C> {
...
-                                                Expr::Object(ObjectLit {
+                                        match expr.as_mut() {
+                                            Expr::Object(ObjectLit { props, .. }) => {
+                                                props.extend(args_props);
+                                            }
+                                            Expr::Call(_) => {
+                                                args_props.push(PropOrSpread::Spread(
+                                                    SpreadElement {
+                                                        dot3_token: DUMMY_SP,
+                                                        expr:expr.to_owned()
```
Comment (kdy1) [inline packages/emotion/transform/src/lib.rs:494]: Same here

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

File: packages/emotion/transform/src/lib.rs (hunk 1 above)
Comment (ImBrek) [reply packages/emotion/transform/src/lib.rs:636]: I am sorry, i don`t know how i can fix it. I am beginner in rust. Can you help me a bit

Comment (ImBrek) [reply packages/emotion/transform/src/lib.rs]: @kdy1 thanks!

**Pull request merged**